daisi/src/manager/core_network.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 The SOLA authors
+//
+// This file is part of DAISI.
+//
+// DAISI is free software: you can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation; version 2.
+//
+// DAISI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+// the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along with DAISI. If not, see
+// <https://www.gnu.org/licenses/>.
+//
+// SPDX-License-Identifier: GPL-2.0-only
+
+#ifndef DAISI_MANAGER_CORE_NETWORK_H_
+#define DAISI_MANAGER_CORE_NETWORK_H_
+
+#include <cstdint>
+#include <vector>
+
+#include "ns3/network-module.h"
+
+namespace ns3 {
+class Ipv4InterfaceContainer;
+class NetDeviceContainer;
+class CsmaHelper;
```

Comment:
Maybe we should add a comment on why those empty declarations are required here?

Reply:
Comment added (forward declarations).